### ISSUE TITLE
iOS Analytics not showing up in status page

### ISSUE THREAD

**Issue by ajincentfit** [Microsoft/appcenter-sdk-cordova#75]
### **Description**

The analytics plugin works as intended on Android, but does not show any users on iOS. The plugin install correctly on iOS (after using the nightly build of cordova-ios that fixes the hardcoded ios 8 versioning on the podfile). The app secret is correctly specified. I get a successful console log from the plugin, but nothing ever shows up online. 

### **Repro Steps**

1. Install analytics plugin for cordova ios
2. Check app center portal for analytics info

### **Details**
Only debug info from the plugin is:
2019-01-29 12:02:30.193482-0500 xxxxxxx[569:280793] [CDVTimer][appcentershared] 0.517011ms
2019-01-29 12:02:30.236199-0500 xxxxxxx[569:280793] [CDVTimer][appcenteranalytics] 42.553902ms
2019-01-29 12:02:30.298250-0500 xxxxxxx[569:280793] [AppCenterCrashes] WARNING: -[MSCrashes configureCrashReporterWithUncaughtExceptionHandlerEnabled:]/590 Detecting crashes is NOT enabled due to running the app with a debugger attached.
2019-01-29 12:02:30.329605-0500 xxxxxxx[569:280793] [CDVTimer][appcentercrashes] 93.222976ms

However, I do get a:
d94ee935 | STARTSERVICE
In log flow for iOS so I know it is syncing correctly with my app secret.

1. Which SDK version are you using?
   0.3.0
2. Which OS version did you experience the issue on?
    iOS 11
3. What version of Cordova, Cordova platforms, Cocoapods, and Ruby are you using?
    - e.g. Cordova version 7.1.0, Cordova iOS 4.5.x branch nightly
4. What device version did you see this error on?  Were you using a simulator or a physical device?
    iPhone 6s and X
5. What third party plugins are you using?
com.cesidiodibenedetto.filechooser 1.0.4 "FileChooser"
com.incentfit.shealth 1.3.0 "Samsung SHealth"
com.lampa.startapp 6.1.6 "startApp"
com.phonegap.plugins.nativesettingsopener 1.0.1 "Native settings"
cordova-android-support-gradle-release 2.0.1 "cordova-android-support-gradle-release"
cordova-background-geolocation-lt 2.14.2 "BackgroundGeolocation"
cordova-plugin-appcenter-analytics 0.3.0 "App Center Analytics for Cordova"
cordova-plugin-appcenter-crashes 0.3.0 "App Center Crashes for Cordova"
cordova-plugin-appcenter-shared 0.3.0 "App Center shared code for Cordova"
cordova-plugin-apprate 1.4.0 "AppRate"
cordova-plugin-background-fetch 5.4.1 "CDVBackgroundFetch"
cordova-plugin-camera 4.0.3 "Camera"
cordova-plugin-cocoalumberjack 0.0.4 "CocoaLumberjack"
cordova-plugin-cocoapod-support 1.6.0 "Cordova CocoaPods Dependency Support"
cordova-plugin-customurlscheme 4.3.0 "Custom URL scheme"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-device-name 1.3.4 "Device-Name"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-geolocation 4.0.1 "Geolocation"
cordova-plugin-globalization 1.11.0 "Globalization"
cordova-plugin-googleplus 5.2.1 "Google SignIn"
cordova-plugin-health 1.1.0 "Cordova Health"
cordova-plugin-inappbrowser 3.1.0-dev "InAppBrowser"
cordova-plugin-ionic 5.2.8 "cordova-plugin-ionic"
cordova-plugin-ionic-webview 2.3.1 "cordova-plugin-ionic-webview"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-network-information 2.0.1 "Network Information"
cordova-plugin-request-location-accuracy 2.2.3 "Request Location Accuracy"
cordova-plugin-splashscreen 5.0.3-dev "Splashscreen"
cordova-plugin-statusbar 2.4.1 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-support-google-services 1.1.0 "cordova-support-google-services"
cordova.plugins.diagnostic 4.0.10 "Diagnostic"
ionic-plugin-keyboard 2.2.1 "Keyboard"
phonegap-plugin-barcodescanner 8.0.0 "BarcodeScanner"
phonegap-plugin-multidex 1.0.0 "Multidex"
phonegap-plugin-push 2.2.3 "PushPlugin"

**Comment by annakocheshkova** [comment]
@ajincentfit yes, this is correct, as per [the guide](https://github.com/Microsoft/appcenter-sdk-cordova#3-configure-preferences). By default this value is `false`.

**Comment by annakocheshkova** [comment]
Do you have any questions/issues left?

**Comment by ajincentfit** [comment]
From my understanding, and the behavior on Android, omitting the preference / making it false should upload analytics data. However, I can only see analytics data when set to true and then enabling in JS. This does work consistently though, so no more issues or questions. Thanks.

**Comment by pburrows** [comment]
If this is the fix PLEASE PLEASE updated the documentation/ readme! It is literally the opposite behavior as what is described:

> APPCENTER_ANALYTICS_ENABLE_IN_JS - (optional, default is "false") This preference controls whether Analytics will be enabled automatically (default option) or will require call to Analytics.setEnabled() in JS code before sending any usage data to App Center portal. This might be useful e.g. in case when an application may want to ask users whether they want to share analytics information.

It says:

* The default option is false
* Analytics are enabled by default.
* Setting it to "true" should turn off the default analytics, and make you have to specify in code whether or not to turn it on.

The word "Enable" in this is enabling the ability to control whether it is on or off in code.

At least, that's what the docs say.
